### PR TITLE
docs: Deprecate `cluster-pool-v2beta`

### PR DESCRIPTION
**Pull request by gandro**
The `cluster-pool-v2beta` IPAM mode has been added in Cilium v1.12 as a preview feature. Since then, we have added a more flexible `multi-pool` IPAM mode (c.f. https://github.com/cilium/cilium/pull/22762), which addresses some of the shortcomings of `cluster-pool-v2`. Mainly, `multi-pool` supports multiple pod CIDR pools, it has a simpler CIDR release mechanism, and it allows users to specify what IP pool a certain pod should allocate from via pod annotation.

Therefore, since `cluster-pool-v2beta` was never made generally available and was never recommended for production (and indeed still has a few known limitations and bugs that make it unlikely it has been deployed in production), we therefore deprecate it in v1.14, with the plan to remove it in v1.15.

The only `cluster-pool-v2beta` feature that is lacking in `multi-pool` IPAM mode is BGP support. We intend to add BGP support to multi-pool IPAM in Cilium v1.15.

This PR also hides a few mentions of the `cluster-pool-v2beta` mode from the command-line help page. This does not impact any functional behavior.

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

**Comment by gandro** [comment]
CI failure looks like https://github.com/cilium/cilium/issues/25785

**Comment by gandro** [comment]
CI was green. Unfortunately this needs a rebase now.

**Comment by gandro** [comment]
/test-1.16-4.19

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -358,6 +358,9 @@ Deprecated Options
 * The ``disable-cnp-status-updates``, ``cnp-node-status-gc-interval duration`` and ``enable-k8s-event-handover``
   options are deprecated and will be removed in v1.15. There is no replacement for these
   flags as enabling them causes scalability and performance issues even in small clusters.
+* The ``cluster-pool-v2beta`` IPAM mode is deprecated and will be removed in v1.15.
+  The functionality to dynamically allocate Pod CIDRs is now provided  by the
+  more flexible ``multi-pool`` IPAM mode.
```
Comment (zacharysarah) [inline Documentation/operations/upgrade.rst:363]: Edits for clarity
```suggestion
  Dynamic allocation of Pod CIDRs is now provided by the
  more flexible ``multi-pool`` IPAM mode.
```

**Review by zacharysarah — APPROVED** [review]
One small change, otherwise LGTM for docs. Approving with the understanding that changes are required prior to merge.

**Pull request merged**